### ISSUE TITLE
PeerJS Server in multiple Instances of EC2

### ISSUE THREAD

**Issue by chattes** [peers/peerjs-server#176]
Hello,
We setup PeerJS server in multiple EC2 instances with NLB distributing load.
But we very quickly realised this will not work as the other Peer will not be able to connect if First Peer has connected in another EC2 instance.

How can we solve this issue, I am thinking of a Global Redis store, but I am not sure how to implement the same.
Any pointers would be highly appreciated.

**Comment by jrowny** [comment]
There's no code for persistence that I can see, maybe we can get a pull request together, it's also a problem if the server ever restarts. 

The best solution I can think of in your situation is to make sure all peers hit the same EC2. Not sure how to make that work with NLB.

**Comment by chattes** [comment]
Yeah ,by taking a cursory look at the Code. I think its just using a Map(). But I maybe wrong.
For now NLB does not seem to be a solution maybe a cluster needs to be built and handled by the app itself.

**Comment by ericwooley** [comment]
I looked into how hard peerjs would be to scale, in case I ever get to that point. 

I believe the easiest option would to add something like redis support here, there would obviously be some refactoring around switching to pubsub. But it might not be so bad.
https://github.com/peers/peerjs-server/blob/12bd0f245e676d6f901a500486c0d51fe3551f0b/src/models/realm.ts#L34 


I have no idea when I will have the need/time to create a PR for this, so if anyone has the time, I would love to see it implemented..
